design/volume-snapshot-data-movement/volume-snapshot-data-movement.md

```diff
@@ -67,7 +66,7 @@ DMs take the responsibility to handle DUCR/DDCRs, Velero provides a built-in DM
 **Node-Agent**: Node-Agent is an existing Velero module that will be used to host VBDM.  
 **Exposer**: Exposer is to expose the snapshot/target volume as a path/device name/endpoint that are recognizable by Velero generic data path. For different snapshot types/snapshot accesses, the Exposer may be different. This isolation guarantees that when we want to support other snapshot types/snapshot accesses, we only need to replace with a new Exposer and keep other components as is.  
 **Velero Generic Data Path (VGDP)**: VGDP is the collective of modules that is introduced in [Unified Repository design][1]. Velero uses these modules to finish data transmission for various purposes. In includes uploaders and the backup repository.  
-**Uploader**: Uploader is the module in VGDP that reads data from the source and writes to backup repository for backup; while read data from backup repository and write to the restore target for restore. At present, only file system uploader is supported. In future, the block level uploader will be added. For file system uploader, only Kopia uploader will be used, Restic will not be integrated with VBDM.   
```

Comment:
We don't need to remove this:
`In future, the block level uploader will be added.`

A block level uploader is required for implementing the block level backup as suggested above for the non-goal. Otherwise, it will be confusing to have a dedicate Uploader layer in the architecture if there is only one uploader.

Reply:
This makes sense, as I think that something to handle changed blocks needs to be in that interface and this one won't ever have that ability. +1

Reply:
updated